internal/cltest/fixtures.go

```diff
@@ -26,18 +26,26 @@ func NewJob() models.JobSpec {
 	return j
 }
 
-func NewTask(taskType string, json ...string) models.TaskSpec {
-	if len(json) == 0 {
-		json = append(json, ``)
+func NewTask(taskType string, args ...interface{}) models.TaskSpec {
+	if len(args) == 0 {
+		args = append(args, ``)
 	}
-	params := JSONFromString(json[0])
+	params := JSONFromString(args[0].(string))
+
 	params, err := params.Add("type", taskType)
 	mustNotErr(err)
 
-	return models.TaskSpec{
+	t := models.TaskSpec{
 		Type:   taskType,
 		Params: params,
 	}
+	if len(args) == 2 {
+		confs := uint64(args[1].(int))
```

Comment:
We've been using variadic arguments as a way to get around operator overloading, but I worry that checking for a length of 2 might be taking it too far. It's not very discoverable to know that the 3rd parameter of `NewTask` takes a confirmation unless you look at the bottom of the implementation. Plus, we miss out on compile time type checking and the like.

May I suggest we add a `NewTaskWithConfirmations` that takes params and confirmations? Open to all sorts of renaming, just want to move away from the variadic `args ...interface{}`.